File: app/src/main/java/com/neome/api/meta/base/dto/StudioEntAutomationStepParseDocument.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdForm;
import com.neome.api.meta.base.Types.MetaIdGrid;
import com.neome.api.meta.base.Types.MetaIdPipelineParam;
import com.neome.api.meta.base.Types.MetaIdVar;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class StudioEntAutomationStepParseDocument extends StudioEntAutomationStep
{
  @Nullable
  public String aiInstructions;

  @Nullable
  public MetaIdField documentFieldId;

  @Nullable
  public Boolean exposeOutputAsPipelineVariable;

  @Nullable
  public MetaIdPipelineParam inputFormPipelineVarId;

  @Nullable
  public StudioValueVarIdCondition iterateOnGridFilterVarId;

  @Nullable
  public MetaIdGrid iterateOnGridId;

  @Nullable
  public MetaIdPipelineParam outputFormPipelineVarId;

  @Nullable
  public MetaIdVar outputMappingVarId;

  @Nullable
  public MetaIdForm schemaFormId;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioEntAutomationStepParseDocument.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdForm
import com.neome.api.meta.base.Types.MetaIdGrid
import com.neome.api.meta.base.Types.MetaIdPipelineParam
import com.neome.api.meta.base.Types.MetaIdVar

class StudioEntAutomationStepParseDocument : StudioEntAutomationStep() {
    var aiInstructions: string? = null
    var documentFieldId: MetaIdField? = null
    var exposeOutputAsPipelineVariable: boolean? = null
    var inputFormPipelineVarId: MetaIdPipelineParam? = null
    var iterateOnGridFilterVarId: StudioValueVarIdCondition? = null
    var iterateOnGridId: MetaIdGrid? = null
    var outputFormPipelineVarId: MetaIdPipelineParam? = null
    var outputMappingVarId: MetaIdVar? = null
    var schemaFormId: MetaIdForm? = null
}
